[WIP] ci: AppVeyor: remove MINGW_64-gcov

blueyed opened the pull request:
There appears to be no locking support on Windows.

This results in a) incomplete results in general, and more impotantly b) those
errors being spilled into test results, e.g.:

    [  FAILED  ] C:/projects/neovim/test/functional\core\main_spec.lua @ 97: Command-line option -s errors out when trying to use nonexistent file with -s
    C:/projects/neovim/test/functional\core\main_spec.lua:98: Expected objects to be the same.
    Passed in:
    (string) 'Cannot open for reading: "Xtest-functional-core-main-s.nonexistent": no such file or directory
    profiling:C:\projects\neovim\build/src/nvim/CMakeFiles/nvim.dir/buffer.c.gcda:Data file mismatch - some data files may have been concurrently updated without locking support
    '
    Expected:
    (string) 'Cannot open for reading: "Xtest-functional-core-main-s.nonexistent": no such file or directory
    '

    stack traceback:
            C:/projects/neovim/test/functional\core\main_spec.lua:98: in function <C:/projects/neovim/test/functional\core\main_spec.lua:97>

Therefore it does not make sense to have a always-failing job.

Just for reference, the locking appears to have been reworked for gcc
9.1 [1], so might could be tried later again.

1: https://github.com/gcc-mirror/gcc/commit/56621355b

-

Followup to 9a9d9a187, where it was ignored.

justinmk commented:
gcov always fails on appveyor. Example: https://ci.appveyor.com/project/neovim/neovim/builds/25164128

blueyed committed:
ci: AppVeyor: remove MINGW_64-gcov

There appears to be no locking support on Windows.

This results in a) incomplete results in general, and more impotantly b) those
errors being spilled into test results, e.g.:

    [  FAILED  ] C:/projects/neovim/test/functional\core\main_spec.lua @ 97: Command-line option -s errors out when trying to use nonexistent file with -s
    C:/projects/neovim/test/functional\core\main_spec.lua:98: Expected objects to be the same.
    Passed in:
    (string) 'Cannot open for reading: "Xtest-functional-core-main-s.nonexistent": no such file or directory
    profiling:C:\projects\neovim\build/src/nvim/CMakeFiles/nvim.dir/buffer.c.gcda:Data file mismatch - some data files may have been concurrently updated without locking support
    '
    Expected:
    (string) 'Cannot open for reading: "Xtest-functional-core-main-s.nonexistent": no such file or directory
    '

    stack traceback:
            C:/projects/neovim/test/functional\core\main_spec.lua:98: in function <C:/projects/neovim/test/functional\core\main_spec.lua:97>

Therefore it does not make sense to have a always-failing job.

Just for reference, the locking appears to have been reworked for gcc
9.1 [1], so might could be tried later again.

1: https://github.com/gcc-mirror/gcc/commit/56621355b

-

Followup to 9a9d9a187, where it was ignored.

blueyed commented:
Updated the PR to remove the job completely.
I've tried to set `GCOV_ERROR_FILE`, but it appears to still default to stderr (due to it being used not via the gcov tool likely?!).

See the updated description / commit message.

blueyed commented:
This should not cause a decrease in coverage, since it fails already before uploading the codecov in general.

It could be fixed by e.g. ignoring those errors in the tests, which might be good to have in general (i.e. a method ignore some errors on stderr), but have not looked into how easy that would be.

blueyed commented:
An option might also be to set/use `GCOV_PREFIX` / `GCOV_PREFIX_STRIP` for subprocesses, either to just write them somewhere else to be ignored / not interfere, or to collect them from there later also.

blueyed commented:
Trying GCOV_ERROR_FILE again https://github.com/neovim/neovim/pull/10198, with keeping it in `helpers.clear`.

blueyed commented:
https://github.com/neovim/neovim/pull/10198 looks promising, closing this one here for now.
